breaking: Make `Root` a `PathCodec`, remove `Empty`

erikvanoosten opened the pull request:
As suggested by @khajavi, we're making the `Root` constant an alias to `PathCodec.empty` so that it can be used in routes as follows:
```scala
Method.GET / Root -> handler(Response.text("Hello world!"))
```
For consistency the `Empty` constant is removed completely.

### Migration
⚠️ This is a breaking change.
All current usages of `Root` have to be replaced with `Path.root`,
all current usages of `Empty` have to be replaced with `Path.empty`.

Also:
- add examples on how to use root routes in the very first examples so that it is easy to find for new users,
- cleanup documentation start page.

erikvanoosten committed:
Add info about home routes

It took me a lot of time to find out how to express a route to the root path, a lot of friction. Lets make sure this never happens again by making this part of the very first example.

Also: remove ZERO WIDTH NON-JOINER character.

khajavi commented:
Good catch! But I think we should change it to better DSL. The `GET / ""` is somewhat unattractive!
Maybe it is better to change the `Root` definition to an alias for `PatchCodec.empty` instead of `Path.root`.

erikvanoosten committed:
breaking: Make `Root` a PathCodec

... to improve the DSL for root routes.

Also: remove the 'watch mode' section from the docs as watch mode is not intended for starting users.

erikvanoosten committed:
Merge branch 'main' into home-route

erikvanoosten commented:
Alright @khajavi , I like that. I have changed the `Root` constant as you suggested.

@jdegoes this is a breaking change, so I have requested a new review.

erikvanoosten commented:
@khajavi, what is the purpose of `Empty`? (Currently defined as `val Empty: Path = Path.empty`.) Should that be a `PathCodec` as well?

erikvanoosten committed:
Replace `Root` by `Path.root`, `Empty` by `Path.empty`

Remove `Empty`.

erikvanoosten commented:
> @khajavi, what is the purpose of `Empty`? (Currently defined as `val Empty: Path = Path.empty`.) Should that be a `PathCodec` as well?

Since I had to replace all occurrences of `Root` with `Path.root`, I went ahead and replaced `Empty` with `Path.empty` as well. Because I think `Empty` is no longer needed I removed it.

erikvanoosten committed:
Document migration

erikvanoosten committed:
Update documentation

erikvanoosten commented:
Since this is a breaking change, I understand if the others think this going to far. If you ask me, `Method.GET / "" -> handler(Response.text("Hello world!"))` is still a valid option.

erikvanoosten committed:
sbt fmt

khajavi commented on docs/migration/RC6-to-xx.md:
```diff
@@ -0,0 +1,3 @@
+**`Root` and `Empty`**
+- replace `Root` with `Path.root`
+- replace `Empty` with `Path.empty`
```
:heart: for documenting the migration guide

erikvanoosten committed:
Merge branch 'main' into home-route

erikvanoosten committed:
Merge 2798

jdegoes commented:
@erikvanoosten Just need a merge conflict fix. Thanks!

erikvanoosten committed:
Merge branch 'main' into home-route

erikvanoosten commented:
> @erikvanoosten Just need a merge conflict fix. Thanks!

@jdegoes Ready to merge!
